### PR TITLE
Footer and Api calls

### PR DESCRIPTION
**Pull request by shrinathprabhu**
Resolves [AR-2447](https://team-1624093970686.atlassian.net/browse/AR-2447).

## Changes

Please include a summary of the changes and any relavant context not convered in the issue ticket. You can assume that reviewers will begin by reading the description of the linked issue.

## Checklist

- [x] The branch name follows the format: `developer/AR-XXX-issue-name`.
- [ ] The changes have been tested locally.

### CODE REVIEW

File: src/pages/AppUsers.vue [hunk 1]
```diff
@@ -1,24 +1,29 @@
...
+import {
+  fetchAllUsers,
+  searchUsers,
+  fetchAllUserTransactions,
+  fetchMonthlyUsers,
+} from '@/services/gateway.service'
 import chartUtils from '@/utils/chart'
 
-let users = ref([])
+let users: Ref<any[]> = ref([])
```
Comment (ajithranka) [inline src/pages/AppUsers.vue:19]: Create a user type if we don't have one.

File: src/store/index.ts [hunk 2]
```diff
@@ -9,15 +9,19 @@ const debug = process.env.NODE_ENV !== 'production'
 const state = {
   env: 'test',
   smartContractAddress: '',
-  showLoader: false,
+  isLoading: false,
   loadingMessage: '',
```
[the remark this answers is not in the document]
Comment (shrinathprabhu) [reply src/store/index.ts:13]: We can consider this as refactor after beta

File: src/pages/AppConfigure.vue [hunk 3]
```diff
@@ -1,23 +1,86 @@
...
+  store.commit('showLoader', 'Saving app config...')
+  const appConfigRequestBody: AppConfig = store.getters.appConfigRequestBody
+  await updateApp(appConfigRequestBody)
+  await updateSmartContractTransactions(appConfigRequestBody)
+  store.commit('hideLoader')
+}
+
+async function updateSmartContractTransactions(app: AppConfig) {
+  try {
+    store.commit('showLoader', 'Updating app name in smart contract...')
```
[the remark this answers is not in the document]
Comment (shrinathprabhu) [reply src/pages/AppConfigure.vue:39]: Will refactor this when we switch to Pinia from Vuex

File: env.d.ts [hunk 4]
```diff
@@ -1,7 +1,23 @@
...
+  import type { SmartContractRequestParams } from '@/utils/signerMakeTx'
+
+  interface Window {
+    signerMakeTx: (data: SmartContractRequestParams) => Promise<string>
+  }
+}
+
+declare module 'vue-router' {
+  interface RouteMeta {
+    requiresAuth: boolean
```
Comment (ajithranka) [inline env.d.ts:19]: Let's keep this field optional.

```suggestion
    requiresAuth?: boolean
```

**Comment by ajithranka** [comment]
Great work @shrinathprabhu! 👏🏼

**Pull request merged**